pytypes/type_util.py

```diff
@@ -216,15 +216,14 @@ def get_orig_class(obj, default_to__class__=False):
             cls = obj.__class__
         if _typing_3_7 and is_Generic(cls):
             # Workaround for https://github.com/python/typing/issues/658
-            stck = stack()
-            # Searching from index 2 is sufficient: At 0 is get_orig_class, at 1 is the caller.
-            # We assume the caller is not typing._GenericAlias.__call__ which we are after.
-            for line in stck[2:]:
+            frame = currentframe()
+            while frame:
```

Comment:
Should you still skip the first 2 frames?

Reply:
perhaps easiest way to skip the first 2 is writing ``frame = currentframe().f_back.f_back`` I guess (?)

Reply:
Wouldn't that skip checking the last two frames?

Reply:
You added a check for file number: "current line number in Python source code". That makes no sense.

Reply:
I think `currentframe().f_back.f_back` as a starting point makes sense.